ttnn/cpp/ttnn/operations/eltwise/unary/device/unary_composite_op.cpp

```diff
@@ -610,4 +610,37 @@ Tensor _glu(const Tensor& input_a, int32_t dim , const std::optional<MemoryConfi
     return glu_result;
 }
 
+// rpow: y = k**(a) = exp( a**log(k) )
+Tensor _rpow(const Tensor& a, float k, const std::optional<MemoryConfig>& output_mem_config) {
+    TT_ASSERT(k > 0.0, "rpow cannot be calcualted for non-positive numbers");
+    float log_k = logf(k);
+
+    Tensor result = ttnn::multiply(a, log_k);
+    return ttnn::exp(result, false);
+}
+
+using HWFunctionT = std::function<Tensor(const Tensor& y, const std::optional<MemoryConfig>&)>;
+Tensor _make_global_from_hw_impl(HWFunctionT fn, const Tensor& y,  const std::optional<MemoryConfig>& output_mem_config) {
+    TT_FATAL(y.get_legacy_shape().rank() == 4, "Cannot support non-rank 4 Tensor");
+
+    // format to HW
+    Tensor y_hw = reshape(y, 1, 1, y.get_legacy_shape()[2], y.get_legacy_shape()[3] * y.get_legacy_shape()[1] * y.get_legacy_shape()[0]);
+
+    // compute @fn
+    Tensor z_0 = fn(y_hw, output_mem_config);
+    TT_FATAL(y_hw.get_legacy_shape() == z_0.get_legacy_shape(), "shape match");
+    y_hw.deallocate();
+
+    // reformat
+    Tensor z_1 = reshape(z_0, y.get_legacy_shape()[0], y.get_legacy_shape()[1], y.get_legacy_shape()[2], y.get_legacy_shape()[3]);
+    z_0.deallocate();
+
+    return z_1;
+}
+
+// Global Norm
+Tensor _normalize_global(const Tensor& y,  const std::optional<MemoryConfig>& output_mem_config) {
```

Comment:
https://github.com/tenstorrent/tt-metal/blob/953d2c504d9f1d4f5917888f422e10f13f5377c1/tests/tt_eager/python_api_testing/sweep_tests/pytorch_ops.py#L83

https://github.com/tenstorrent/tt-metal/blob/953d2c504d9f1d4f5917888f422e10f13f5377c1/tests/tt_eager/python_api_testing/sweep_tests/pytorch_ops.py#L1447

attach golden function for both ops in unary.py

Reply:
use above links for golden fn  reference